support fast, simple `sg start single-program-experimental-blame-sqs` for local dev

sqs opened the pull request:
This makes it easier to run Sourcegraph in local dev by compiling a few key services (frontend, searcher, repo-updater, gitserver, and worker) into a single Go binary and running that.

Compared to `sg start` (which compiles and runs ~10 services), it's faster to start up (by ~10% or a few seconds), takes a lot less memory and CPU when running, has less log noise, and rebuilds faster. It is slower to recompile for changes just to `frontend` because it needs to link in more code on each recompile, but it's faster for most other Go changes that require recompilation of multiple services.

This is only intended for local dev as a convenience. There may be different behavior in this mode that could result in problems when your code runs in the normal deployment. Usually our e2e tests should catch this, but to be safe, you should run in the usual mode if you are making sensitive cross-service changes.

Partially reverts "svcmain: Simplify service setup (#61903)" (commit 95410322925a4d78883290e01095086d11239877).


## Test plan

Existing tests cover any regressions to existing behavior. This new behavior is for local dev only.

slimsag reviewed (approved):
neat; I wish we could go further with this && make our deployments way simpler

sqs commented:
> neat; I wish we could go further with this && make our deployments way simpler

Yeah -- we absolutely can!

sqs commented:
I renamed it to `single-program-experimental-blame-sqs` and added a disclaimer about it. I’m going to merge it since it makes my own local development significantly smoother and faster. If I can prove that it helps other people too, then I’ll propose to promote it to a non-experimental-blame-sqs way of running it. :P
